### PR TITLE
feat(xLoad): Add xLoad binding capabilities

### LINKED ISSUE #925: Multiple limitations on x:Load
## I'm submitting a...

Feature request

## Current behavior

 Currently `x:Load` in Uno:
 - doesn't support binding
 - doesn't support being toggled multiple times
 - is only applied if the Visibility is also set/bound in XAML (bug)

## Expected behavior

Supports the above

## Minimal reproduction of the problem with instructions

```xaml
<StackPanel Background="Yellow" Width="200">
			<CheckBox x:Name="LoadCheckBox" Margin="30" IsChecked="{x:Bind IsLoad, Mode=TwoWay}"/>
			<Border x:Name="LoadBorder" Width="50" Height="50" Background="Green" x:Load="{x:Bind IsLoad, Mode=OneWay}"/>
		</StackPanel>
```
Code-behind:
```chsharp
		public bool IsLoad
		{
			get { return (bool)GetValue(IsLoadProperty); }
			set { SetValue(IsLoadProperty, value); }
		}

		// Using a DependencyProperty as the backing store for IsLoad.  This enables animation, styling, binding, etc...
		public static readonly DependencyProperty IsLoadProperty =
			DependencyProperty.Register("IsLoad", typeof(bool), typeof(MainPage), new PropertyMetadata(false));
```

## Environment

Nuget Package: Uno.UI

Package Version(s): 1.45.0-dev.1560

Affected platform(s):
- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): Closes #925, fixes https://github.com/unoplatform/uno/issues/5068

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?

Currently x:Load in Uno:

- doesn't support binding
- doesn't support being toggled multiple times
- is only applied if the Visibility is also set/bound in XAML (bug)

## What is the new behavior?

The above scenarios are supported, based on the work from @kazo0.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 1]
```diff
@@ -4998,20 +5025,38 @@ private IDisposable TryGenerateDeferedLoadStrategy(IIndentedStringBuilder writer
...
 					var hasDataContextMarkup = dataContextMember != null && HasMarkupExtension(dataContextMember);
 
-					var disposable = writer.BlockInvariant($"new {XamlConstants.Types.ElementStub}()");
+					var currentOwnerName = CurrentResourceOwnerName;
+
+					var elementStubHolderNameStatement = "";
+
+					if (HasImplicitViewPinning)
+					{
+						// Build the ElemenStub builder holder variable to ensute that the element stub
```
Comment (kazo0) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:5041]:
```suggestion
						// Build the ElementStub builder holder variable to ensure that the element stub
```

:D

File: src/Uno.UI.FluentTheme/themeresources.xaml [hunk 2]
```diff
@@ -17283,6 +17283,8 @@
                 <VisualState x:Name="Photo">
                   <VisualState.Setters>
                     <Setter Target="PersonPictureEllipse.Fill" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=TemplateSettings.ActualImageBrush}" />
+                    <!-- ImageBrush is not stretching properly https://github.com/unoplatform/uno/issues/4951 -->
+                    <wasm:Setter Target="PersonPictureImage.Source" Value="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=TemplateSettings.ActualImageBrush.ImageSource}" />
```
Comment (kazo0) [inline src/Uno.UI.FluentTheme/themeresources.xaml:17287]: Should this be in this PR?

File: src/Uno.UI/UI/Xaml/ElementStub.cs [hunk 3]
```diff
@@ -15,12 +18,65 @@
...
+
+		// Using a DependencyProperty as the backing store for Load.  This enables animation, styling, binding, etc...
+		public static readonly DependencyProperty LoadProperty =
+			DependencyProperty.Register("Load", typeof(bool), typeof(ElementStub), new PropertyMetadata(
+				false, OnLoadChanged));
+
+		public ElementStub(Func<View> contentBuilder) : this()
+		{
+#if UNO_HAS_UIELEMENT_IMPLICIT_PINNING
+			// In this context, the delegate provided here closes over other UIElement intances
```
Comment (kazo0) [inline src/Uno.UI/UI/Xaml/ElementStub.cs:47]:
```suggestion
			// In this context, the delegate provided here closes over other UIElement instances
```

File: src/Uno.UI/UI/Xaml/ElementStub.cs [hunk 4]
```diff
@@ -15,12 +18,65 @@
...
+			// causing memory leaks unless the Target member of the delegate is weak.
+			// Here, we deconstruct the delegate to keep the MethodInfo, and invoking it
+			// via the resolution of the weak reference. This technique only works because
+			// the provided delegate and its target (the lambda's display class) is kept
+			// alive by the "ElementSub holder" variable provided by the XAML generator.
```
Comment (kazo0) [inline src/Uno.UI/UI/Xaml/ElementStub.cs:52]:
```suggestion
			// alive by the "ElementStub holder" variable provided by the XAML generator.
```

**Review by kazo0 — APPROVED** [review]

File: src/Uno.UI.FluentTheme/themeresources.xaml (hunk 2 above)
Comment (jeromelaban) [reply src/Uno.UI.FluentTheme/themeresources.xaml:17287]: Yes and no. I'll move it to another PR.

File: src/Uno.UI.RuntimeTests/Uno.UI.RuntimeTests.csproj [hunk 5]
```diff
@@ -102,6 +102,12 @@
 	<ItemGroup>
 	  <Compile Include="Tests\Windows_UI_Xaml_Controls\HtmlElementAttributeTests\Given_HtmlElementAttribute.Wasm.cs" />
 	</ItemGroup>
+	
+	<ItemGroup>
+	  <Page Update="Tests\Windows_UI_Xaml\Controls\xLoad_xBind.xaml">
```
Comment (davidjohnoliver) [inline src/Uno.UI.RuntimeTests/Uno.UI.RuntimeTests.csproj:107]: don't merge

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**